pkg/imagehandler/imagehandler.go

```diff
@@ -63,20 +66,46 @@ func (f *imageFileSystem) getBaseImage(initramfs bool) baseFile {
 	}
 }
 
-func (f *imageFileSystem) ServeImage(name string, ignitionContent []byte, initramfs bool) (string, error) {
+func (f *imageFileSystem) getNameForKey(key string) (name string, err error) {
+	if img, exists := f.images[key]; exists {
+		return img.name, nil
+	}
+	rand, err := uuid.NewRandom()
+	if err == nil {
+		name = rand.String()
+	}
```

Comment:
under what conditions would we have to rely on the random name? is it really needed?

Reply:
I have mixed feelings about it, but the point is that we're serving the images for all namespaces from one endpoint, that is quite possibly accessible to everyone in the cluster. Random names avoid leaking information across namespaces.